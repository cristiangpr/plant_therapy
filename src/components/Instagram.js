import React, { Component } from 'react';


import Instafeed from "instafeed.js";
class Instagram extends Component {




  componentDidMount(){
    var feed = new Instafeed({
      accessToken: process.env.REACT_APP_INSTA_TOKEN,
        get: 'user',
        clientId: 	process.env.REACT_APP_INSTA_CLIENT,
         template:'<div class="col-lg-3  col-md-6   wow animated zoomIn "><a class="instagram-image" href="{{link}}" target="_blank"><img src="{{image}}"/> <div className="likes" id="likes">&hearts; {{likes}}</div></a></div>',
        limit: 4,

        resolution: 'low_resolution', // thumbnail, low_resolution, standard_resolution
        sortBy: 'most-recent', // none, least-commented, least-liked, least-recent, most-commented, most-liked, most-recent, random
        tagName: null,
        userId: 14251104739,
  });
  feed.run();
  }

  render(){
    return (

<>

      <div className="row" id="instafeed"></div>


</>
  );
  };
};
export default Instagram;
