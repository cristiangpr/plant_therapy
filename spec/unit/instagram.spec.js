import React from 'react';
import Instagram from "../../src/components/Instagram.js";
 import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';


describe(Instagram, function() {
  it('renders correctly', () => {
    const tree = renderer
      .create(
          <MemoryRouter>
            <Instagram/>
        </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
