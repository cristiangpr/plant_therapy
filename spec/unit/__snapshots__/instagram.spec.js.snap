// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Instagram renders correctly 1`] = `
<section
  className="blog-area section-gap"
  id="blog"
>
  <div
    className="container"
  >
    <div
      className="row justify-content-center"
    >
      <div
        className="col-lg-12"
      >
        <div
          className="section-header"
        >
          <h3>
            INSTAGRAM
          </h3>
        </div>
      </div>
    </div>
    <div
      className="row"
      id="instafeed"
    />
  </div>
</section>
`;
